### PR TITLE
feat: pay-for-order support w/ tokenized cart PRBs

### LINKED ISSUE #8530: PRB tokenized cart on pay-for-order flow
```[tasklist]
### Compatibility
- [x] Ensure compatibility with "Product Add-Ons"
- [x] Ensure compatibility with "WooCommerce Deposits"
- [x] Ensure compatibility with "WooCommerce Bookings"
- [x] Ensure compatibility with base product types: simple, variable, downloadable, subscriptions, virtual products
```

```[tasklist]
### Functionality
- [x] On PRB click, customer can check out with items on the cart
- [x] Price, tax, shipping, line items are displayed accurately
- [x] Order page reflects selected product and paid amount
```

### PR DESCRIPTION
**Pull request by frosso**
Fixes https://github.com/Automattic/woocommerce-payments/issues/8530

#### Changes proposed in this Pull Request

Adding support of the pay-for-order page w/ the tokenized cart feature.
I created a separate `PaymentRequestOrderApi` class to interact with the store, because the API is different than the checkout API.

#### Testing instructions

- Enable the "Enable Cart-Token implementation for PRBs" flag in your dev tools (you'll need to update them)
- Ensure you have Google Pay/Apple Pay enabled in the merchant's settings
- As a merchant, create an order in the backend with some items to pay for
	- PLEASE NOTE: the cart amount needs to be greater than $50 USD - investigating here: p1718364553103609-slack-CU6SYV31A
	- PLEASE NOTE: the order data needs to have an email address associated to it for the Store API to work properly, investigating here: p1718365667412379-slack-C02TS23QJ1X
- Once the order is saved, click on the "Customer payment page" link (you can use this link both in incognito or as an anonymous customer)
- Click on the Google Pay/Apple Pay button displayed on the page
- You should be able to select a payment method (but not shipping options)
- Place the order
- The charge for the order should match the order's total amount

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8946` or branch name `feat/tokenized-cart-pay-for-order-flow` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=feat/tokenized-cart-pay-for-order-flow) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: c69958668a94cfa8b2ad9f3c8c8f520eb618a66e
- Build time: 2024-06-14 12:08:16 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/tokenized-payment-request/payment-request.js [hunk 1]
```diff
@@ -139,93 +139,104 @@ export default class WooPaymentsPaymentRequest {
...
-					! _self.isPaymentAborted &&
-					didItemsNeedShipping === newCartData.needs_shipping
-				) {
-					paymentRequest.update( {
-						total: {
-							label: getPaymentRequestData( 'total_label' ),
-							amount: parseInt(
-								newCartData.totals.total_price,
-								10
+		if ( getPaymentRequestData( 'button_context' ) === 'pay_for_order' ) {
```
Comment (frosso) [inline client/tokenized-payment-request/payment-request.js:142]: There's a lot of indentation changes here - using https://github.com/Automattic/woocommerce-payments/pull/8946/files?w=1 to hide whitespace changes helps a lot

File: client/tokenized-payment-request/transformers/wc-to-stripe.js [hunk 2]
```diff
@@ -47,6 +47,14 @@ export const transformCartDataForDisplayItems = ( cartData ) => {
 		} );
 	}
 
+	const refundAmount = parseInt( cartData.totals.total_refund || '0', 10 );
```
Comment (frosso) [inline client/tokenized-payment-request/transformers/wc-to-stripe.js:50]: Sneaky change in chase there are refund line items

File: client/tokenized-payment-request/order-api.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+		} );
+	}
+
+	/**
+	 * Returns the customer's order object.
+	 * See https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/StoreApi/docs/cart.md#get-cart
+	 *
+	 * @return {Promise} Cart response object.
+	 */
+	async getCart() {
```
Comment (frosso) [inline client/tokenized-payment-request/order-api.js:66]: I initially called this method `getOrder`, but then I thought of renaming it `getCart`. After all, it's returning a cart response object. And renaming it slightly simplifies the branching logic 6e4d8980b

**Review by brettshumaker — APPROVED** [review]
This looks good. I don't see anything in these changes that wouldn't be picking up the payment method correctly (the issue you mentioned in slack about a bnpl requirement being enforced on a Google Pay transaction).

**Pull request merged**